GVFS/GVFS.Notifications/VFSForGit.Mac/StatusMenuItem/IPC/VFSForGitNotification.m

```diff
@@ -0,0 +1,142 @@
+#import "VFSForGitNotification.h"
+
+NSString const * IdentifierKey = @"Id";
+NSString const * EnlistmentKey = @"Enlistment";
+NSString const * EnlistmentCountKey = @"EnlistmentCount";
+NSString const * TitleKey = @"Title";
+NSString const * MessageKey = @"Message";
+
+NSString const * AutomountTitle = @"GVFS AutoMount";
+NSString const * AutomountStartMessageFormat = @"Attempting to mount %d GVFS repos(s)";
+NSString const * AutomountSuccessMessageFormat = @"The following GVFS repo is now mounted: \n%@";
+NSString const * AutomountFailureMessageFormat = @"The following GVFS repo failed to mount: \n%@";
+
+@interface VFSForGitNotification()
+
+@property (readwrite, nonnull) NSString *title;
+@property (readwrite, nonnull) NSString *message;
+
+@end
+
+@implementation VFSForGitNotification
+
++ (BOOL) tryCreateNotification:(VFSForGitNotification *_Nonnull *_Nonnull) notification
+                          from:(NSDictionary *_Nonnull) jsonMessage
+                         error:(NSString *_Nonnull *_Nonnull) error
+{
+    NSParameterAssert(notification);
```

Comment:
I haven't used NSParameterAssert() but it seems like this will throw an exception and trigger a termination of the app. Are you sure all the different instances you have added so far, are all crash-worthy and non-recoverable conditions? In a lot of cases, nil parameters may be ignorable, like in the case of output error parameters if the caller doesn't care about the error then it should be fine to fail without the output error.

Reply:
Makes sense. Have removed the hard requirement for output error args. There are 4 other places, where `NSParameterAssert` is used. These are there to catch programmer errors (like for example `VFSMessageParser::tryParseData` invoked with `nil` arg for capturing the output  `parsedMessage`). I think it would be useful to rather crash and collect the logs, if it happens.

Reply:
Also - I've retained the ```_Nonnull``` specifier for errors (even though there is no `NSParameterAssert` for them.) It helps in showing a compiler warning when passing `nil` for errors.

Reply:
Update - changed the types of all errors to NSError and also using the default nullability specifiers for NSError type.